pkg/rancher-desktop/components/Preferences/BodyWsl.vue

```diff
@@ -1,47 +1,84 @@
 <script lang="ts">
 
 import Vue from 'vue';
-import { mapGetters } from 'vuex';
+import { mapGetters, mapState } from 'vuex';
 
+import RdTabbed from '@pkg/components/Tabbed/RdTabbed.vue';
+import Tab from '@pkg/components/Tabbed/Tab.vue';
 import WslIntegration from '@pkg/components/WSLIntegration.vue';
-import RdFieldset from '@pkg/components/form/RdFieldset.vue';
+import WslProxy from '@pkg/components/WSLProxy.vue';
 import { Settings } from '@pkg/config/settings';
-import { RecursiveTypes } from '@pkg/utils/typeUtils';
+import { TransientSettings } from '@pkg/config/transientSettings';
+import { ServerState } from '@pkg/main/credentialServer/httpCredentialHelperServer';
+import { RecursivePartial } from '@pkg/utils/typeUtils';
 
 import type { PropType } from 'vue';
 
 export default Vue.extend({
   name:       'preferences-body-wsl',
-  components: { WslIntegration, RdFieldset },
-  props:      {
+  components: {
+    WslIntegration, WslProxy, RdTabbed, Tab,
+  },
+  props: {
     preferences: {
       type:     Object as PropType<Settings>,
       required: true,
     },
   },
-  computed: { ...mapGetters('preferences', ['getWslIntegrations']) },
-  methods:  {
-    onChange(distro: string, value: boolean) {
-      const property: keyof RecursiveTypes<Settings> = `WSL.integrations["${ distro }"]` as any;
-
-      this.$store.dispatch('preferences/updateWslIntegrations', { distribution: `["${ distro }"]`, value });
-      this.$store.dispatch('preferences/updatePreferencesData', { property, value });
+  computed: {
+    ...mapGetters('transientSettings', ['getActiveTab']),
+    ...mapState('credentials', ['credentials']),
+    activeTab(): string {
+      return this.getActiveTab || 'integration';
```

Comment:
You will need to add the `integration` tab as a new entry into the `preferencesNavItems` object.
It defines the navigator's structure for the preferences dialog and it is also used to validate the `TransientSettings` payload in our Rest API requests:

https://github.com/rancher-sandbox/rancher-desktop/blob/4d8f2bf1c5162b9486ebf4c14f085368f4d2be59/pkg/rancher-desktop/window/preferences.ts#L18-L29

Reply:
@torchiaf I've been looking to solve the failing e2e test 

My changes modified the ``` is the following:

```typescript
export const defaultTransientSettings = {
  noModalDialogs: false,
  preferences:    {
    navItem: {
      current:     'Application' as NavItemName,
      currentTabs: {
        Application:        'general',
        'Container Engine': 'general',
        WSL:                'integration',
      } as Record<NavItemName, string | undefined>,
    },
  },
};
```

When running the failing test 

```typescript
test('setting existing transient settings should be a no-op', async() => {
  let resp = await doRequest('/v1/transient_settings');
  const rawSettings = resp.body.read().toString();
                                                                           
  resp = await doRequest('/v1/transient_settings', rawSettings, 'PUT');
  expect({
    ok:     resp.ok,
    status: resp.status,
    body:   resp.body.read().toString(),
  }).toEqual({
    ok:     true,
    status: 202,
    body:   expect.stringContaining('No changes necessary'),
  });
});
```

It will get the `defaultTransientSettings` object and compare it to the `preferencesNavItems` object.

```typescript
export const preferencesNavItems: NavItems[] = [
  {
    name: 'Application',
    tabs: ['general', 'behavior', 'environment'],
  },
  {
    name: process.platform === 'win32' ? 'WSL' : 'Virtual Machine',
    tabs: process.platform === 'win32' ? ['integration', 'proxy'] : [],
  },
  {
    name: 'Container Engine',
    tabs: ['general', 'allowed-images'],
  },
  { name: 'Kubernetes' },
];
```

Because the test are done on Linux the `{"WSL": "integration"}` tab does not exist.

Do you have any pointer on how to solve this ? It seems that the usecase of having "tabbed page" for specific platform is not taken into account yet.

Reply:
I guess adding the condition `process.platform === 'win32'` also into `defaultTransientSettings` would solve the issue:
```
export const defaultTransientSettings = {
  noModalDialogs: false,
  preferences:    {
    navItem: {
      current:     'Application' as NavItemName,
      currentTabs: {
        Application:        'general',
        'Container Engine': 'general',
        ...(process.platform === 'win32' && { WSL: 'integration' }),
      } as Record<NavItemName, string | undefined>,
    },
  },
};
```

Reply:
@torchiaf Thanks, I haven't thought about using the spread operator to run a condition directly on the key. This should work in theory.